auth/proxy/tests/unit/firebaseJwt.unit.test.ts

```diff
@@ -148,6 +149,22 @@ describe("firebaseJwt", () => {
         })).rejects.toThrow(JsonWebTokenError);
     })
 
+    it("should throw an error when Algorithm is incorrect", async () => {
+        (decode as Mock).mockReturnValueOnce({
+            sub: "mocked-app-id",
+            exp: Math.floor(Date.now() / 1000) + 3600,
+            header: {
+                kid: "mocked-kid",
+                alg: "Test-Algorithm", // Invalid algorithm
```

Comment:
Nitpick - we could parameterize this test and run different algorithms we support i.e.
```ts
it.each([
  "None",
  "HS256"
])
(...)
```